helm-service/chart/templates/_helpers.tpl

```diff
@@ -69,7 +69,7 @@ Return the proper image name
 {{- define "helm-service.images.image" -}}
 {{- $global := .context.Values.global -}}
 {{- $registryName := .imageRoot.registry -}}
-{{- $repositoryName := .imageRoot.repository -}}
+{{- $repositoryName := .imageRoot.repositoryImageName -}}
```

Comment:
Are the renames necessary?
What do they do?

Reply:
Turns out, that Keptn needs to be upgraded with the Helm flag `--reuse-values` set to true. This is also what the cli command `keptn upgrade` (which is just a wrapper around helm) does. Using this flag and the changes of your PR ultimately causes [this](https://github.com/keptn/keptn/issues/5248) bug. A simple "fix" was just to rename that variable to something else, which does not yet exist. While this worked for the core helm chart it did not work for the charts of helm-service and jmeter-service, for whatever reason

Reply:
Those other charts have their own copies of the same templates, which would need to be identically modified.
I suggested in #4933 to move common template helpers into their own [library chart](https://helm.sh/docs/topics/library_charts/) to ease maintenance & to preclude issues like this.
That would involve releasing an additional chart and updating this repo's release process to include that.
Since I don't know that process, I'm not about to do that myself.

Most of the template logic follows that I found in the [Bitnami common library chart](https://github.com/bitnami/charts/tree/master/bitnami/common).

This bit of code
https://github.com/keptn/keptn/blob/63c4e09d35ff2ff5f9e5f04753dda283d12bf8de/helm-service/chart/templates/_helpers.tpl#L79-L83
attempts to anticipate the issue this PR tries to fix.
In default `values.yaml`, I might explore not setting registry or setting the registry to empty either globally or for each image perhaps as a default.
The default docker repository is already `docker.io` & that may not need to be explicit.

Reply:
Will #5456 suffice?

Reply:
thx, i will test and come back to you :) thank you very much for the insides (no helm pro here :-p )